src/components/timeline/index.tsx

```diff
@@ -0,0 +1,288 @@
+import React, {useCallback, useMemo, useEffect, useState} from 'react';
+import {StyleSheet, ImageRequireSource, ViewStyle, MeasureOnSuccessCallback, LayoutChangeEvent} from 'react-native';
+import {Colors, Spacings} from '../../style';
+import View from '../view';
+import Icon from '../icon';
+import Text from '../text';
+import Dash from './Dash';
+
+
+const LINE_WIDTH = 2;
+const POINT_SIZE = 12;
+const HALO_WIDTH = 4;
+const HALO_TINT = 70;
+const HOLLO_WIDTH = 2;
+const CONTENT_POINT_SIZE = 20;
+const ICON_SIZE = 12;
+const CONTENT_CONTAINER_PADDINGS = Spacings.s2;
+const POINT_MARGINS = Spacings.s1;
+
+export enum StateTypes {
+  CURRENT = 'current', // default
+  NEXT = 'next',
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+export enum LineTypes {
+  FULL = 'full', // default
+  DASHED = 'dashed'
+}
+
+export enum PointTypes {
+  BULLET = 'bullet', // default
+  HOLLOW = 'hollow',
+  HALO = 'halo'
+}
+
+export type LineProps = {
+  state?: StateTypes;
+  type?: LineTypes;
+  color?: string;
+  /** to mark as entry point */
+  entry?: boolean;
+}
+
+export type PointProps = {
+  state?: StateTypes;
+  type?: PointTypes;
+  color?: string;
+  icon?: ImageRequireSource;
+  label?: number;
+  /** to align point to this view's center */
+  alignmentTargetRef?: React.MutableRefObject<undefined>;
+  /** the target view's top parent view */
+  targetContainerRef?: React.MutableRefObject<undefined>;
+}
+
+type Position = {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+export type TimelineProps = {
+  height: number;
```

Comment:
I don't like it much either but we need it for the Dashed line and the target alignment. I can keep looking for a better solution but it will delay the release

Reply:
That's fine, I don't think it'll be ready for today release anyway. 
If you'd like we can go over it together. I'll be happy to help find an alternative solution..